File: ui/types/art-style.ts
// 艺术风格状态类型
export type ArtStyleStatus = 'active' | 'inactive';

// 艺术风格数据接口
export interface ArtStyle {
  name: string;          // 风格名称
  image: string;         // 风格示例图片URL
  keyword: string;       // 生成图片时使用的关键词
  status: ArtStyleStatus;// 状态
  created_at: string;    // 创建时间
  updated_at: string;    // 更新时间
}

// 获取艺术风格列表的请求参数
export interface GetArtStylesParams {
  status?: ArtStyleStatus;
}

// 获取艺术风格列表的响应数据
export interface GetArtStylesResponse {
  art_styles: ArtStyle[];
  total: number;
} 

File: backend/models/art_style.py
from datetime import datetime
from typing import Optional
from beanie import Document, Indexed

class ArtStyle(Document):
    """艺术风格模型
    用于存储角色和背景的艺术风格类型
    """
    name: Indexed(str, unique=True)  # 风格名称
    image: str  # 风格示例图片URL
    keyword: str  # 生成图片时使用的关键词
    status: str = "active"  # 状态：active/inactive
    created_at: datetime = datetime.utcnow()
    updated_at: datetime = datetime.utcnow()

    class Settings:
        name = "art_styles"  # 集合名称
        
    class Config:
        schema_extra = {
            "example": {
                "name": "Heavy Impasto",
                "image": "https://example.com/image.png",
                "keyword": "Heavy Impasto",
                "status": "active"
            }
        } 